### ISSUE TITLE
Our var of CUDA_SEPARABLE_COMPILATION should be BLT_ CUDA_SEPARABLE_COMPILATION

### ISSUE THREAD

**Issue by cyrush** [LLNL/blt#282]
CUDA_SEPARABLE_COMPILATION is the per target prop name, 
CMAKE_CUDA_SEPARABLE_COMPILATION is the global setting

but we are using:
CUDA_SEPARABLE_COMPILATION in the blt macros

**Comment by mclarsen** [comment]
On the subject of CUDA related variable, should there be a `BLT_CUDA_ARCH` variable that does the right thing? I have seen multiple ways to set this.

**Comment by mclarsen** [comment]
Adding `BLT_` to the cmake variables that BLT handles would  help users understand what BLT is doing and what it is not.

**Comment by cyrush** [comment]
dejavu  @mclarsen

**Comment by adayton1** [comment]
I agree that the naming is a little confusing. I recently noticed that we are passing -DCUDA_SEPARABLE_COMPILATION to our dependencies that use BLT, but it seems like we should be using -DCMAKE_CUDA_SEPARABLE_COMPILATION instead, though I'm not sure what BLT is actually expecting.
